pkg/agent/nodeportlocal/k8s/npl_controller.go

```diff
@@ -389,34 +400,45 @@ func (c *NPLController) handleAddUpdatePod(key string, obj interface{}) error {
 	var err error
 	var updatePodAnnotation bool
 	var nodePort int
-	newPodPorts := make(map[int]struct{})
-	newPodContainers := newPod.Spec.Containers
-	for _, container := range newPodContainers {
+	podPorts := make(map[int]struct{})
+	podContainers := pod.Spec.Containers
+	nplAnnotations := []NPLAnnotation{}
+
+	podAnnotation, nplExists := pod.GetAnnotations()[NPLAnnotationKey]
+	if nplExists {
+		if err := json.Unmarshal([]byte(podAnnotation), &nplAnnotations); err != nil {
+			klog.Warningf("Unable to unmarshal NodePortLocal annotation for Pod %s/%s",
+				pod.Namespace, pod.Name)
```

Comment:
for consistency, you can use `key` directly in log messages to print `<podNamespace>/<podName>`

Reply:
Changed. Thanks.